lib/faraday/deprecate.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module Faraday
+  # @param new_klass [Class] new Klass to use
+  #
+  # @return [Class] A modified version of new_klass that warns on
+  #   usage about deprecation.
+  # @see Faraday::Deprecate
+  module DeprecatedClass
+    def self.proxy_class(new_klass)
```

Comment:
I think this should accept a `version` argument so we can proxy deprecated classes beyond 1.0. Probably not necessary for this PR on the `0.1x` branch, but definitely once this gets ported to `master`.